### PR TITLE
Use a different message when writing tags fails due to permissions issues, add test

### PR DESCRIPTION
**Pull request by adreed-msft**
If the failure point includes data about tags, we now treat that as a failure writing tags, and log about needing tags permissions (per stgexp's request).

A single test case covers Upload & S2S cases for writing tags, as well as different blob types, hitting all code paths.

### CODE REVIEW

**Review by tanyasethi-msft — APPROVED** [review]
Looks good to me, the pipelines seem to fail because of the MaxStorageAccountsCountPerSubscriptionExceeded, waiting for the cleanup, along with a string linter error which might need attention.

**Comment by seanmcc-msft** [comment]
Related https://github.com/Azure/azure-storage-azcopy/issues/2798

**Review by vibhansa-msft — APPROVED** [review]

File: ste/sender-blobSymlinks.go [hunk 1]
```diff
@@ -68,16 +68,33 @@ func (s *blobSymlinkSender) SendSymlink(linkData string) error {
...
 			HTTPHeaders:  &s.headersToApply,
 			Metadata:     s.metadataToApply,
 			Tier:         s.destBlobTier,
-			Tags:         s.blobTagsToApply,
+			Tags:         blobTags,
 			CPKInfo:      s.jptm.CpkInfo(),
 			CPKScopeInfo: s.jptm.CpkScopeInfo(),
 		})
-	return err
+	if err != nil {
```
Comment (gapra-msft) [inline ste/sender-blobSymlinks.go:86]: was this a bug - us not doing a separate set tags in certain situations for symlinks? Should we add an explicit symlink test?

File: ste/sender-blobSymlinks.go [hunk 2]
```diff
@@ -68,16 +68,33 @@ func (s *blobSymlinkSender) SendSymlink(linkData string) error {
...
+		s.jptm.FailActiveSend(common.Iff(len(blobTags) > 0, "Upload symlink (with tags)", "Upload symlink"), err)
```
Comment (gapra-msft) [inline ste/sender-blobSymlinks.go:87]: I think we need to also make this change with sender-blobFolders

File: ste/sender-blobSymlinks.go (hunk 1 above)
Comment (adreed-msft) [reply ste/sender-blobSymlinks.go:86]: Yes, it appears so. I'll port this to folders and modify the tests.

File: ste/sender-blobFolders.go [hunk 3]
```diff
@@ -199,17 +199,33 @@ func (b *blobFolderSender) EnsureFolderExists() error {
...
 			&blockblob.UploadOptions{
 				HTTPHeaders:  &b.headersToApply,
 				Metadata:     b.metadataToApply,
-				Tags:         b.blobTagsToApply,
+				Tags:         blobTags,
 				CPKInfo:      b.jptm.CpkInfo(),
 				CPKScopeInfo: b.jptm.CpkScopeInfo(),
 			})
+		if err != nil {
+			b.jptm.FailActiveSend(common.Iff(len(blobTags) > 0, "Upload symlink (with tags)", "Upload symlink"), err)
```
Comment (gapra-msft) [inline ste/sender-blobFolders.go:218]:
```suggestion
			b.jptm.FailActiveSend(common.Iff(len(blobTags) > 0, "Upload folder (with tags)", "Upload folder"), err)
```

**Review by gapra-msft — APPROVED** [review]

**Pull request merged**